Python 3.14 Backports (Dec. 29, 2025)

MingcongBai opened the pull request:
Topic Description
-----------------

- pyzmq: move cython to build dep; improve PKGDES
- autobuild4: update to 4.14.0
    Co-authored-by: 白铭骢 \(Mingcong Bai\) \(@MingcongBai\) <jeffbai@aosc.io>
- libsigrok: fix build with SWIG 4.4
    Track patches at AOSC-Tracking/libsigrok @ aosc/v0.5.2+git20241021
    \(HEAD: 86644efc04e2c9948c270d11e262fbef79e515be\).
- lirc: move stray /lib => /usr/lib
- elinks: use Lua 5.4
- pyzmq: drop unneeded ABSPLITDBG=0
- pyzmq: update to 26.4.0
- libnvme: directly depend on python-3
- libsigrok: update to 0.5.2+git20241021
- future: update to 1.0.0
    - Drop Python 2 bindings.
    - Improve PKGDES.
    - Use Git tag.

... and 23 more commits

Package(s) Affected
-------------------

- autobuild4: 4.14.0
- breezy: 3.3.12-2
- cleo: 2.2.1-1
- coherent.build: 0.35.2
- coherent.licensed: 0.5.2
- dulwich: 0.24.8-1
- elinks: 1:0.19.0
- future: 1.0.0
- jaraco.classes: 3.4.0
- jaraco.context: 6.0.1
- jaraco.functools: 4.2.1
- keyring: 25.6.0
- libnvme: 1.16-1
- libsignon-glib: 2.1-5
- libsigrok: 0.5.2+git20241021
- lirc: 0.10.2
- more-itertools: 10.7.0
- platformdirs: 4.3.8
- pygame-sdl2: 8.3.7.25031702-2
- pyinotify: 0.0.1
- pyzmq: 26.4.0
- tomlkit: 0.13.3
- toolbelt: 1.0.0-3
- virt-install: 5.1.0-1
- virt-manager: 5.1.0-1
- yubikey-manager: 5.8.0-1

Security Update?
----------------

No

Build Order
-----------

```
#buildit autobuild4 coherent.build:+stage2 libsignon-glib elinks yubikey-manager virt-manager pyinotify lirc breezy toolbelt more-itertools jaraco.functools jaraco.context jaraco.classes coherent.licensed coherent.build keyring cleo platformdirs tomlkit dulwich pygame-sdl2 future libsigrok libnvme pyzmq
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`
- [x] LoongArch 64-bit (No SIMD) `loongarch64_nosimd`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

xtexx committed:
pyzmq: update to 26.4.0

xtexx committed:
libnvme: directly depend on python-3

xtexx committed:
libsigrok: update to 0.5.2+git20241021

xtexx committed:
future: update to 1.0.0

- Drop Python 2 bindings.
- Improve PKGDES.
- Use Git tag.

xtexx committed:
pygame-sdl2: fix header installation

xtexx committed:
dulwich: fix deps and SRCS

xtexx committed:
tomlkit: use pypi sources

xtexx committed:
platformdirs: update to 4.3.8

xtexx committed:
keyring: update to 25.6.0

xtexx committed:
cleo: remove unused dependencies; update description

xtexx committed:
coherent.build: new, 0.35.2

xtexx committed:
coherent.licensed: new, 0.5.2

xtexx committed:
jaraco.classes: new, 3.4.0

xtexx committed:
jaraco.context: new, 6.0.1

xtexx committed:
more-itertools: new, 10.7.0

xtexx committed:
jaraco.functools: new, 4.2.1

xtexx committed:
toolbelt: use pypi sources; update description

xtexx committed:
breezy: skip PyO3 ABI compatibility check

xtexx committed:
lirc: update to 0.10.2

xtexx committed:
pyinotify: switch to elephant fork

The original upstream stopped maintenance ten years ago.
This fork removed asyncore usage in code and thus works on Python 3.12+.

xtexx committed:
virt-manager: patch to use lxml instead of libxml2

xtexx committed:
virt-manager: byte-compile Python code

MutsukiC committed:
yubikey-manager: remove unused setuptools dependency

Co-authored-by: xtex <xtex@aosc.io>

chenx97 committed:
elinks: drop use of orphaned dblatex

chenx97 committed:
libsignon-glib: drop use of orphaned dbus-test-runner

MingcongBai committed:
libsignon-glib: disable gtk-doc and broken tests

../tests/check_signon.c: In function ‘test_auth_session_creation_fn’:
../tests/check_signon.c:391:28: error: too many arguments to function ‘signon_identity_new’; expected 0, have 2
  391 |     SignonIdentity *idty = signon_identity_new(NULL, NULL);
      |                            ^~~~~~~~~~~~~~~~~~~ ~~~~

MingcongBai committed:
elinks: update to 0.19.0

- Disable broken Ruby bindings.
- Keep only man pages.
- Track patches at AOSC-Tracking/elinks @ aosc/v0.19.0
  (HEAD: 01a907b94ef3e433602196e866ffa751ec1390a3).

MingcongBai committed:
pyzmq: drop unneeded ABSPLITDBG=0

MingcongBai committed:
elinks: use Lua 5.4

MingcongBai committed:
lirc: move stray /lib => /usr/lib

xtexx commented on lang-python/pyzmq/autobuild/defines:
```diff
@@ -1,6 +1,8 @@
 PKGNAME=pyzmq
 PKGSEC=python
-PKGDEP="zeromq python-3"
+PKGDEP="zeromq python-3 cython"
```
`cython` should be a build dep, maybe?

MingcongBai committed:
libsigrok: fix build with SWIG 4.4

Track patches at AOSC-Tracking/libsigrok @ aosc/v0.5.2+git20241021
(HEAD: 86644efc04e2c9948c270d11e262fbef79e515be).

aosc-buildit-bot committed:
autobuild4: update to 4.14.0

Co-authored-by: 白铭骢 (Mingcong Bai) (@MingcongBai) <jeffbai@aosc.io>

MingcongBai committed:
pyzmq: move cython to build dep; improve PKGDES

xtexx reviewed (approved):
Reviewed-by
